File: sentil-matlab/examples/offline_monitoring.m
% Offline robustness over a recorded trace, in discrete and dense time.

trace = sentil.Trace([0 1 2 3 4], 'speed', [12 9 7 4 6]);
phi = sentil.Formula.parse('always (speed > 5)');

fprintf('robustness: %g\n', phi.robustness(trace));
fprintf('per sample: %s\n', mat2str(phi.robustness_signal(trace)));

spans = phi.violations(trace);
fprintf('violations:');
for i = 1:size(spans, 1)
    fprintf(' [%g, %g]', spans(i, 1), spans(i, 2));
end
fprintf('\n');

fprintf('dense robustness: %g\n', phi.robustness_dense(trace));

File: sentil-core/examples/offline_monitoring.rs
//! Offline robustness over a recorded trace, in discrete and dense time.
//!
//! Run with `cargo run --example offline_monitoring`.

use sentil::{Formula, Trace};

fn main() -> sentil::Result<()> {
    let mut trace = Trace::new(vec![0.0, 1.0, 2.0, 3.0, 4.0])?;
    trace.add_signal("speed", vec![12.0, 9.0, 7.0, 4.0, 6.0])?;

    let phi = Formula::parse("always (speed > 5)")?;
    println!("robustness:       {}", phi.robustness(&trace)?);
    println!("per sample:       {:?}", phi.robustness_signal(&trace)?);
    println!("dense robustness: {}", phi.robustness_dense(&trace)?);
    Ok(())
}